### PR TITLE
Cppcheck errors isolated

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

This is the code changes I made when addressing cppcheck warnings/issues (PR pending) that are likely to create diffs. Putting them in a separate PR to isolate them.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by jmarrec** [comment]
Some diffs in the `RetailPackagedTESCoil.idf`, as expected I guess. It occurs only on the summer design day, right when the mode is about to change from DischargeOnly = 5 (which is the mode where I made the change) to CoolingOnly = 1.

This is for `PSZ-AC:2_COOLC DXCOIL`


image

File: src/EnergyPlus/FanCoilUnits.cc [hunk 1]
```diff
@@ -3550,6 +3550,7 @@ namespace FanCoilUnits {
                                FanCoil(FanCoilNum).HCoilType_Num != HCoil::Electric) {
                         state.dataLoopNodes->Node(WaterControlNode).MassFlowRate = PLR * FanCoil(FanCoilNum).MaxHeatCoilFluidFlow;
                     }
+                    Calc4PipeFanCoil(state, FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOut, PLR);
```
Comment (Myoldmopar) [inline src/EnergyPlus/FanCoilUnits.cc:3553]: Yep, this was just fully missing.  Nice.

File: src/EnergyPlus/OutputReportTabular.cc [hunk 2]
```diff
@@ -15705,12 +15706,15 @@ void WriteLoadComponentSummaryTables(EnergyPlusData &state)
...
+
                         airLoopHeatTable.peakDesSensLoad = -finalSysSizing.SysHeatCoinSpaceSens;
                         airLoopHeatTable.designPeakLoad = -finalSysSizing.SysDesHeatLoad;
+
                         airLoopCoolTable.diffPeakEst = airLoopCoolTable.peakDesSensLoad - airLoopCoolTable.estInstDelSensLoad;
                         airLoopCoolTable.diffDesignPeak = airLoopCoolTable.designPeakLoad - airLoopCoolTable.peakDesSensLoad;
+
                         airLoopHeatTable.diffPeakEst = airLoopHeatTable.peakDesSensLoad - airLoopHeatTable.estInstDelSensLoad;
-                        airLoopCoolTable.diffDesignPeak = airLoopHeatTable.designPeakLoad - airLoopHeatTable.peakDesSensLoad;
+                        airLoopHeatTable.diffDesignPeak = airLoopHeatTable.designPeakLoad - airLoopHeatTable.peakDesSensLoad;
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportTabular.cc:15717]: Nice.  Yep.

File: src/EnergyPlus/OutputReportTabular.cc [hunk 3]
```diff
@@ -13985,6 +13985,7 @@ void WritePredefinedTables(EnergyPlusData &state)
             for (mUnqObjNames = 1; mUnqObjNames <= numUnqObjName; ++mUnqObjNames) {
                 if (curObjectName == uniqueObjectName(mUnqObjNames)) {
                     found = mUnqObjNames;
+                    break;
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportTabular.cc:13988]: Yeah might as well break early here :+1:

File: src/EnergyPlus/PackagedThermalStorageCoil.cc [hunk 4]
```diff
@@ -4589,7 +4589,7 @@ void CalcTESCoilDischargeOnlyMode(EnergyPlusData &state, int const TESCoilNum, R
             ElecCoolingPower = TotCap * EIR * RuntimeFraction;
         }
         // now see if coil is running dry
-        PartLoadOutAirEnth = EvapInletEnthalpy - (TotCap * PartLoadRatio) / EvapAirMassFlow;
+        PartLoadOutAirEnth = EvapInletEnthalpy - (TotCap * PLR) / EvapAirMassFlow;
```
Comment (Myoldmopar) [inline src/EnergyPlus/PackagedThermalStorageCoil.cc:4592]: How did you find this?

File: src/EnergyPlus/OutdoorAirUnit.cc [hunk 5]
```diff
@@ -2502,10 +2502,10 @@ namespace OutdoorAirUnit {
             case CompType::UnitarySystemModel: { // 'CompType:UnitarySystem'
                 if (Sim) {
                     // This may have to be done in the unitary system object since there can be both cooling and heating
-                    if (((OpMode == Operation::NeutralMode) && (OutAirUnit(OAUnitNum).controlType == OAUnitCtrlType::Temperature)) &&
+                    if (((OpMode == Operation::NeutralMode) && (OutAirUnit(OAUnitNum).controlType == OAUnitCtrlType::Temperature)) ||
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutdoorAirUnit.cc:2505]: Thanks CppCheck

**Review by Myoldmopar — APPROVED** [review]
OK this is a great set of isolated changes that could trigger diffs.  Thank you for keeping them separate from the other bulk changes.  I don't have issue with them, but I am wondering about the PartLoadRatio -> PLR change.  Was CppCheck complaining about something there?

**Comment by Myoldmopar** [comment]
Diffs seem isolated and reasonable.  I don't think it's worth another round of CI.  And I just read the commit log explaining why we should use the updated PLR value which was recalculated if needed.  OK, this is done.  Merging.  Thanks @jmarrec

File: src/EnergyPlus/PackagedThermalStorageCoil.cc (hunk 4 above)
Comment (jmarrec) [reply src/EnergyPlus/PackagedThermalStorageCoil.cc:4592]: I can't recall if there was a specific warning. I may just have noticed this while reducing scope of variables and const-ing stuff. 🤷

**Pull request merged**